src/main/java/io/electrum/prepaidutility/api/IEventsResource.java

```diff
@@ -0,0 +1,27 @@
+package io.electrum.prepaidutility.api;
+
+import io.electrum.prepaidutility.model.PurchaseResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.ServerErrorException;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+public interface IEventsResource {
+
+   default void notifyTokenPurchase(
+         String purchaseId,
+         PurchaseResponse body,
+         SecurityContext securityContext,
+         AsyncResponse asyncResponse,
+         Request request,
+         HttpServletRequest httpServletRequest,
+         HttpHeaders httpHeaders,
+         UriInfo uriInfo) {
+
+      asyncResponse.resume(new ServerErrorException("This operation has not been implemented.", 501));
```

Comment:
```suggestion
      throw new UnsupportedOperationException("This operation has not been implemented.");
```

Both of these will result in an HTTP 501 being returned to the client. But calling `asyncResponse.resume` forces that response without the implementing class having the opportunity to intercept it.

I think throwing the exception provides better transparency to the implementer who can catch it and handle it in some other way if desired.

I made [the same comment](https://github.com/electrumpayments/airtime-service-interface/pull/61#discussion_r481814688) for @gevashkar-rampersadh who's adding a new operation to the Airtime API. I seem to be outnumbered now though so perhaps I need to consider I'm wrong 😨 

Thoughts?

Reply:
I think it let's stick with it as-is. Like @gevashkar-rampersadh said, the implementer of `IEventsResource` has control either way and the original implementation's effect is a bit clearer to any reader. What you say does apply to EventsResource, but we are not really expecting any interception there are we?

Reply:
I was thinking more about cases where a new method is provided with a `default` implementation. If someone had already implemented the interface before the `default` method existed they'd never know to intercept the new method. But I think an implementer is more likely to spot that something's changed if they're catching exceptions so throwing an exception rather than actually returning an HTTP 501 on behalf of the implementer would make the change more obvious.

Well, that was my thinking but I'm less convinced now.